Source/items.cpp

```diff
@@ -516,50 +514,23 @@ void CalcSelfItems(Player &player)
 
 bool GetItemSpace(Point position, int8_t inum)
 {
-	int xx = 0;
-	int yy = 0;
+	Point xy_candidates[9];
+	int candidate_count = 0;
 	for (int j = position.y - 1; j <= position.y + 1; j++) {
-		xx = 0;
 		for (int i = position.x - 1; i <= position.x + 1; i++) {
-			itemhold[xx][yy] = ItemSpaceOk({ i, j });
-			xx++;
-		}
-		yy++;
-	}
-
-	bool savail = false;
-	for (int j = 0; j < 3; j++) {
-		for (int i = 0; i < 3; i++) { // NOLINT(modernize-loop-convert)
-			if (itemhold[i][j])
-				savail = true;
+			if (ItemSpaceOk({ i, j }))
+				xy_candidates[candidate_count++] = { i, j };
 		}
 	}
 
 	int rs = GenerateRnd(15) + 1;
 
-	if (!savail)
+	if (candidate_count == 0)
 		return false;
 
-	xx = 0;
-	yy = 0;
-	while (rs > 0) {
-		if (itemhold[xx][yy])
-			rs--;
-		if (rs <= 0)
-			continue;
-		xx++;
-		if (xx != 3)
-			continue;
-		xx = 0;
-		yy++;
-		if (yy == 3)
-			yy = 0;
-	}
-
-	xx += position.x - 1;
-	yy += position.y - 1;
-	Items[inum].position = { xx, yy };
-	dItem[xx][yy] = inum + 1;
+	Point &destination_xy = xy_candidates[rs % candidate_count];

```

Comment:
Removing the loop changes the way `rs` is used in a way that requires removing the +1 for correct results. At this point the expected range for rs is currently 1-15, since the range doesn't start at 0 the first potential position is considered less frequently than it was previously.

Reply:
This is also probably why the tests are failing, items are being dropped in the next candidate space in the search list than when the demo was recorded.